lib/spack/spack/test/util/path.py

```diff
@@ -134,3 +134,32 @@ def test_path_debug_padded_filter(debug, monkeypatch):
     monkeypatch.setattr(tty, "_debug", debug)
     with spack.config.override("config:install_tree", {"padded_length": 128}):
         assert expected == sup.debug_padded_filter(string)
+
+
+spack_root = os.environ["SPACK_ROOT"]
+
+
+@pytest.mark.parametrize(
+    "path,expected",
+    [
+        ("/home/spack/path/to/file.txt", "/home/spack/path/to/file.txt"),
+        ("file:///home/another/config.yaml", "/home/another/config.yaml"),
+        ("path/to.txt", f"{os.sep.join([spack_root, 'path', 'to.txt'])}"),
+    ],
+)
+def test_canonicalize_linux_file(path, expected):
```

Comment:
Can we join these tests and just use conditional parameterization? Or, if that's clunky/unreadable (personally I'm on the fence), if one of these tests isn't supposed to run on a given platform, that we skip it.

Reply:
@tldahlgren For this PR, you can disregard this comment